src/herbivoresAndCarnivores.js

```diff
@@ -2,14 +2,42 @@
 
 class Animal {
   // write your code here
+  static alive = [];
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
   // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
+
+  unhide() {
+    this.hidden = false;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(herbivore) {
+    if (herbivore.hidden === false && herbivore.health > 0) {
```

Comment:
The logic in the `bite` method should be enhanced to ensure it doesn't proceed if the target is another carnivore. Currently, it only checks if the herbivore is hiding, but it should also verify the type of the target animal.